modules/GenebassGeneBurden.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python
+"""This module extracts and processes target/disease evidence from the Genebass webservice."""
+
+import argparse
+import logging
+import sys
+
+from pyspark.sql import SparkSession
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.functions import coalesce, col, lit, log10, pow, round, when
+from pyspark.sql.types import IntegerType
+
+from common.evidence import initialize_sparksession, read_trait_mappings, write_evidence_strings
+
+METHOD_DESC = {
+    'pLoF': 'Burden test carried out with rare pLOF variants.',
+    'missense|LC': 'Burden test carried out with rare missense variants including low-confidence pLOF and in-frame indels.',
+    'synonymous': 'Burden test carried out with rare synonymous variants.',
+    'pLoF|missense|LC': 'Burden test carried out with pLOF or missense variants.',
+}
+
+
+def main(genebass_data: str, genebass_trait_mappings: str, spark_instance: SparkSession) -> DataFrame:
+    """
+    This module extracts and processes target/disease evidence from the raw Genebass Portal.
+    """
+    logging.info(f'File with the Genebass gene burden results: {genebass_data}')
+    logging.info(f'File with the Genebass traits with their EFO mappings: {genebass_trait_mappings}')
+
+    # Load data
+    genebass_trait_mappings_df = read_trait_mappings(
+        genebass_trait_mappings, study_name='Genebass', spark_instance=spark_instance
+    )
+    logging.info(f'{genebass_trait_mappings_df.count()} Genebass trait mappings have been loaded.')
+
+    genebass_df = (
+        spark_instance.read.parquet(genebass_data)
+        .filter(col('Pvalue_Burden') <= 6.7e-7)
+        .select(
+            'gene_id',
+            'annotation',
+            'n_cases',
+            'n_controls',
+            'trait_type',
+            'phenocode',
+            'description',
+            'Pvalue_Burden',
+            'BETA_Burden',
+            'SE_Burden',
+        )
+        # Joining step with the mappings
+        .join(
+            genebass_trait_mappings_df.withColumnRenamed('PROPERTY_VALUE', 'description'),
+            on='description',
+            how='left',
+        )
+        .distinct()
+        .repartition(20)
+        .persist()
+    )
+
+    # Write output
+    evd_df = parse_genebass_evidence(genebass_df)
+
+    if evd_df.filter(col('resourceScore') == 0).count() != 0:
+        logging.error('There are evidence with a P value of 0.')
+        raise AssertionError(
+            f"There are {evd_df.filter(col('resourceScore') == 0).count()} evidence with a P value of 0."
+        )
+    if not 8_000 < evd_df.count() < 9_000:
```

Comment:
Can we assume this resource getting bigger? If so the upper boundary might be too stringent.

Reply:
We can. As you say, this is an expectation I _currently_ have but which might change. Basically I prefer to pick up an increase when generating the evidence than when we later check the metrics. However I agree that runtime is not the place to do this quality checks